datacube/index/memory/_products.py

```diff
@@ -19,8 +20,9 @@
 
 
 class ProductResource(AbstractProductResource):
-    def __init__(self, metadata_type_resource):
-        self.metadata_type_resource: MetadataTypeResource = metadata_type_resource
+    def __init__(self, index):
+        self._index = index
+        self.metadata_type_resource: MetadataTypeResource = index.metadata_types
```

Comment:
why have this when it seems we're now using `self._index.metadata_types` instead? If there is a difference between the two and their use cases, it should be clarified

Reply:
Yeah, just me being lazy.  I'll fix it.